signature/internal/rekor_set.go

```diff
@@ -40,29 +41,30 @@ type UntrustedRekorPayload struct {
 // A compile-time check that UntrustedRekorSET implements json.Unmarshaler
 var _ json.Unmarshaler = (*UntrustedRekorSET)(nil)
 
-// UnmarshalJSON implements the json.Unmarshaler interface
-func (s *UntrustedRekorSET) UnmarshalJSON(data []byte) error {
-	err := s.strictUnmarshalJSON(data)
-	if err != nil {
-		if formatErr, ok := err.(JSONFormatError); ok {
-			err = NewInvalidSignatureError(formatErr.Error())
-		}
+// ConvertFormatError converts JSONFormatError to NewInvalidSignatureError.
+// All other errors are returned as is.
+func ConvertFormatError(err error) error {
+	var formatErr JSONFormatError
+	if errors.As(err, &formatErr) {
+		return NewInvalidSignatureError(formatErr.Error())
 	}
 	return err
 }
 
-// strictUnmarshalJSON is UnmarshalJSON, except that it may return the internal JSONFormatError error type.
-// Splitting it into a separate function allows us to do the JSONFormatError → InvalidSignatureError in a single place, the caller.
```

Comment:
When `untrustedSignature` and `UntrustedSigstorePayload` continue to use a separate `strictUnmarshalJSON`, I’d rather keep the pattern consistent for this high-risk data.

Reply:
Done that in #2518.